src/ynEIGEN/ynEigenViewer.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+
+import {IERC20, IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import {IynEigen} from "../interfaces/IynEigen.sol";
+import {ITokenStakingNodesManager,ITokenStakingNode} from "../interfaces/ITokenStakingNodesManager.sol";
+
+import {AssetRegistry} from "./AssetRegistry.sol";
+
+contract ynEigenViewer {
+    
+    struct AssetInfo {
+        address asset;
+        string name;
+        string symbol;
+        uint256 ratioOfTotalAssets;
+        uint256 totalBalance;
+    }
+    
+    AssetRegistry public assetRegistry;
```

Comment:
make these immutable so we can deploy this as a proxy

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/145/commits/5de937eb1b24988a0dd41055c952704cb9f80b52